Modernize

loic opened the pull request.

marten-de-vries committed:
Makes _users and _replicator databases system databases (sets extra protections for them like in CouchDB). No regressions, but JS test coverage isn't complete yet.

marten-de-vries committed:
JS is all at 1.0.0 at least now & published.

marten-de-vries committed:
Plug-in improvements

marten-de-vries committed:
All kinds of stuff

marten-de-vries committed:
Upped some npm versions

marten-de-vries committed:
bugfix

marten-de-vries committed:
1.0.4; moved to github

loic committed:
Upgrade pouchdb-plugin-helpers so we test against PouchDB ^6.2.0.

loic committed:
Bump pouchdb-req-http-query to ^2.0.0.

Needed to avoid https://github.com/pouchdb/pouchdb-req-http-query/issues/18.

loic committed:
Removed duplicated dependency.

loic committed:
Moved pouchdb-validation/index.js to lib subfolder as per monorepo convention.

loic committed:
Add 'packages/node_modules/pouchdb-system-db/' from commit '987c0efb9fdd0c7f538a917befb6b20f799032f2'

git-subtree-dir: packages/node_modules/pouchdb-system-db
git-subtree-mainline: dc7dbc3ccf66eec9864485e15953e28db194f108
git-subtree-split: 987c0efb9fdd0c7f538a917befb6b20f799032f2

loic committed:
Adjust pouchdb-system-db for the monorepo.

loic committed:
Update npm on Travis so it supports relative path dependencies.

loic committed:
Update package.json to refer to the local sub-projects instead of their released version.

loic commented on package.json:
```diff
@@ -43,32 +42,34 @@
     "nomnom": "^1.8.1",
     "object-assign": "^4.1.0",
     "on-finished": "^2.3.0",
+    "pouchdb": "newlogic42/dist-pouchdb",
```
`pouchdb` build that includes the https://github.com/pouchdb/pouchdb/pull/6793 fix.

marten-de-vries commented:
Great that you managed to get the tests to pass with more packages in the monorepo, and with a modern version of PouchDB. I think that in the long term, we could probably get rid of all the ``file:...`` dependencies and take the same approach as for the pouchdb repo by moving ``/packages/node_modules`` to ``/node_modules``. That said, I'm all for merging this into pouchdb-server as soon as we can fix the 'linking to an unreleased pouchdb' problem.

loic commented:
@marten-de-vries do you see any downside to the `file:...` approach? From where I stand it describes precisely what's required to build the monorepo.

Did you get any feedback re the beta release hosted on npm?

marten-de-vries commented:
> @marten-de-vries do you see any downside to the file:... approach? From where I stand it describes precisely what's required to build the monorepo.

@loic It seems fine to me, just pointing out that long-term it might be less work to have exactly the same configuration for both repos. It would prevent having to duplicate maintenance work on e.g. the automated publish script.

---

> Did you get any feedback re the beta release hosted on npm?

No, I haven't got an IRC response yet. I thought I'd look through the release code and then just try it for myself, but noticed I'm not personally an npm author for all the packages (especially, pouchdb). Time to try it this way:

@daleharvey @willholley @garrensmith (as the most recent pouchdb releasers): is running 'BETA=1 npm run release' a valid strategy to give @loic a patched pouchdb dependency for pouchdb-server without inconveniencing others? If so, could one of you do so for the latest PouchDB master? Thanks!

loic commented:
Dear all, gentle reminder.

marten-de-vries commented:
Sorry, I haven't had the time to follow through on this. Anyway, with PouchDB's recent 6.4.0 this should be unblocked.

loic commented:
Thanks @marten-de-vries, I plan to resume this effort in the next couple of days.

loic committed:
Remove duplicated dependencies.

loic commented:
@marten-de-vries @gr2m created a new PR against `pouchdb-server`: https://github.com/pouchdb/pouchdb-server/pull/274

That should be enough to get a running build against PouchDB 6.4.0. Once this is merged I'm planning to focus on merging the remaining dependencies to the monorepo.

loic commented:
Reopening for CI.

loic committed:
Upgraded PouchDB to 6.4.1.

Reordered package.json so that packages that belong (or should belong) to the
same monorepo are adjacent. Hopefully this makes it more obvious that they
should be upgraded in lockstep.

gr2m commented:
I’m all for moving forward with this :)

loic committed:
Tweak the express-pouchdb mode to work with the latest PouchDB test suite.
